src/environment.go

```diff
@@ -65,11 +66,15 @@ type environmentInfo interface {
 }
 
 type environment struct {
-	args     *args
-	cwd      string
-	commands map[string]string
+	args *args
+	cwd  string
 }
 
+var (
+	commands map[string]string = make(map[string]string)
```

Comment:
is there a specific reason we need to move it outside the `environment` scope?

Reply:
I tried by just adding the lock but the issue was still there. Honestly, I have to investigate. I don't know if it was because the variable was initialized elsewhere and the lock here. I'll dig to see the reason behind.